Source/App.js

```diff
@@ -1 +1,417 @@
+
+    //Load STK World Terrain
+    viewer.terrainProvider = new Cesium.CesiumTerrainProvider({
+        url : 'https://assets.agi.com/stk-terrain/world',
+        requestWaterMask : true, // required for water effects
+        requestVertexNormals : true // required for terrain lighting
+    });
+    //Enable depth testing so things behind the terrain disappear.
+    viewer.scene.globe.depthTestAgainstTerrain = true;
+
+    //////////////////////////////////////////////////////////////////////////
+    // Loading Data
+    //////////////////////////////////////////////////////////////////////////
+
+    var geojsonOptions = {
+        markerSize : 40,
+        markerSymbol : '!',
+        clampToGround : true
+    };
+    // Load points of interest from a GeoJson
+    var pointsPromise = Cesium.GeoJsonDataSource.load('./Source/SampleData/pointsOfInterest.geojson', geojsonOptions);
+
+
+    var kmlOptions = {
+        camera : viewer.scene.camera,
+        canvas : viewer.scene.canvas
+    };
+    // Load neighborhood boundaries from KML file
+    var neighborhoodsPromise = Cesium.KmlDataSource.load('./Source/SampleData/neighborhoods.kml', kmlOptions);
+
+    //////////////////////////////////////////////////////////////////////////
+    // Styling Data
+    //////////////////////////////////////////////////////////////////////////
+
+    var distanceDisplayCondition = new Cesium.DistanceDisplayCondition(10.0, 10000.0);
+    var neighborhoods = viewer.entities.add(new Cesium.Entity());
+    neighborhoodsPromise.then(function(dataSource) {
+        // Add the new data as entities to the viewer
+        viewer.dataSources.add(dataSource);
+
+        // Get the array of entities
+        var entities = dataSource.entities.values;
+
+        for (var i = 0; i < entities.length; i++) {
+            var entity = entities[i];
+            var name = entity.id;
+
+            if (Cesium.defined(entity.polygon)) {
+                //Set the polygon material to a random, translucent color.
+                entity.polygon.material = Cesium.Color.fromRandom({
+                    red : 0.1,
+                    maximumGreen : 0.5,
+                    minimumBlue : 0.5,
+                    alpha : 0.6
+                });
+                // Make the polygon conform to terrain.
+                entity.polygon.height = undefined;
+                entity.polygon.heightReference = Cesium.HeightReference.CLAMP_TO_GROUND;
+                // Add to the neighborhoods group
+                entity.parent = neighborhoods;
+                // Generate Polygon center
+                var poly_center = Cesium.BoundingSphere.fromPoints(entity.polygon.hierarchy.getValue().positions).center;
+                poly_center = Cesium.Ellipsoid.WGS84.scaleToGeodeticSurface(poly_center);
+                entity.position = poly_center;
+                // Generate labels
+                entity.label = new Cesium.LabelGraphics({
```

Comment:
I would just set label equal to the in the object instead of explicitly creating a new `LabelGraphics`